### PR TITLE
DOC: fix typos

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

**Comment by mdhaber** [comment]
@seberg do we change old release notes? And what's the advice for typo-only PRs?

**Comment by rkern** [comment]
Typo-only PRs are [welcomed, but might be low priority](https://numpy.org/doc/stable/dev/howto-docs.html#contributing-fixes).

File: doc/source/release/2.0.0-notes.rst [hunk 1]
```diff
@@ -395,7 +395,7 @@ replaced with ``NPY_RAVEL_AXIS``. See also :ref:`migration_maxdims`.
 ``NPY_MAXARGS`` not constant and ``PyArrayMultiIterObject`` size change
 -----------------------------------------------------------------------
 Since ``NPY_MAXARGS`` was increased, it is now a runtime constant and not
-compiletime constant anymore.
+compile time constant anymore.
```
Comment (mdhaber) [inline doc/source/release/2.0.0-notes.rst:398]: If we're going to change this, consider hyphenating (as a compound modifier) for clarity.
```suggestion
compile-time constant anymore.
```

**Review by mdhaber — COMMENTED** [review]
It seems that everything being changed was a mistake, and I don't see any objective new mistakes. Before merging, I'd like to get clarification on the project's preference for modifying old release notes.

**Comment by mattip** [comment]
> clarification on the project's preference for modifying old release notes

We have no stated policy, but such fixes are really really low priority. If it is easier to accept the PR than to constantly close PRs, then maybe just accept the change.

File: doc/source/release/2.0.0-notes.rst (hunk 1 above)
Comment (rouault) [reply doc/source/release/2.0.0-notes.rst]: done

**Pull request merged**